File: src/main/scala/com/uwsoft/editor/renderer/SceneLoader.scala
package com.uwsoft.editor.renderer

import com.badlogic.gdx.utils.viewport.Viewport
import com.uwsoft.editor.renderer.resources.ResourceManager

import scala.scalajs.js
import scala.scalajs.js.annotation.JSName


@JSName("uwsoft.editor.renderer.SceneLoader")
@js.native
class SceneLoader extends js.Object {

    var engine:Engine = js.native

    def getRm(): ResourceManager = js.native

    def loadScene(name: String, viewport: Viewport): Unit = js.native
    def loadVoFromLibrary(name: String): Any = js.native

}

File: src/main/js/uwsoft/editor/renderer/SceneLoader.js
/**
 * @JSName("uwsoft.editor.renderer.SceneLoader")
 */
import Engine from 'uwsoft/editor/renderer/Engine'
import ResourceManager from 'uwsoft/editor/renderer/resources/ResourceManager'

export default class SceneLoader {
    constructor(name, viewport) {
        this.name = ''
        this.viewport = null
        this.engine = new Engine()
        this.rm = new ResourceManager()
        this.rm.initAllResources()
        this.sceneVO = null
    }
    loadScene(sceneName, viewport) {
        this.name = name
        this.viewport = viewport
        this.sceneVO = this.rm.getSceneVO(sceneName)
        return this.sceneVO

    }
    getRm() {return this.rm;}
    loadVoFromLibrary(libraryName) {
        let projectInfoVO = this.getRm().getProjectVO()
        let compositeItemVO = projectInfoVO.libraryItems[libraryName]
        return compositeItemVO
    }
    
}

